cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java

```diff
@@ -348,7 +348,9 @@ public ProgramLiveInfo getLiveInfo(Id.Program program, ProgramType type) {
 
     public ClusterResourceReporter(MetricsCollectionService metricsCollectionService, Configuration hConf,
                                    CConfiguration cConf) {
-      super(metricsCollectionService.getCollector(ImmutableMap.of(Constants.Metrics.Tag.CLUSTER_METRICS, "true")));
+      super(metricsCollectionService.getCollector(
+        ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, Constants.SYSTEM_NAMESPACE,
```

Comment:
Yes, I know. I just added these tags explicitly in addition to using system if it does not exist. The ``MetricRecordsWrapper`` has the condition to add a 'system' namespace if it doesn't exist as well. ``MetricHandlerTest`` has tests to demonstrate.